### PR TITLE
matchms scores formatter: added key parameter

### LINKED ISSUE #318: matchms scores formatter: error due to missing key
@maximskorik seems like we currently rely on the name of the molecule being stored as `compound_name` - which might not always be the case for mgf's

```
Traceback (most recent call last):
  File "/mnt/volume/shared/galaxy/var/shed_tools/toolshed.g2.bx.psu.edu/repos/recetox/matchms_formatter/f9c4f477ae91/matchms_formatter/formatter.py", line 139, in <module>
    cli(obj={})
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1665, in invoke
    super().invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/mnt/volume/shared/galaxy/var/shed_tools/toolshed.g2.bx.psu.edu/repos/recetox/matchms_formatter/f9c4f477ae91/matchms_formatter/formatter.py", line 111, in cli
    ctx.obj['data'] = load_data(scores_filename)
  File "/mnt/volume/shared/galaxy/var/shed_tools/toolshed.g2.bx.psu.edu/repos/recetox/matchms_formatter/f9c4f477ae91/matchms_formatter/formatter.py", line 96, in load_data
    scores, matches = scores_to_dataframes(scores)
  File "/mnt/volume/shared/galaxy/var/shed_tools/toolshed.g2.bx.psu.edu/repos/recetox/matchms_formatter/f9c4f477ae91/matchms_formatter/formatter.py", line 77, in scores_to_dataframes
    query_names = [spectra.metadata['compound_name'] for spectra in scores.queries]
  File "/mnt/volume/shared/galaxy/var/shed_tools/toolshed.g2.bx.psu.edu/repos/recetox/matchms_formatter/f9c4f477ae91/matchms_formatter/formatter.py", line 77, in <listcomp>
    query_names = [spectra.metadata['compound_name'] for spectra in scores.queries]
KeyError: 'compound_name'
```

TLDR

- [ ] introduce a `key` parameter to the tool form which is `compound_name` by default
- [ ] replace the `compound_name` string in the `formatter.py` script with a variable and extract this variable up to the top layer where it gets passed as a parameter
- [ ] --> Use this key in the tool to access the query and reference spectra and to add this value into the output table
- [ ] Implement a galaxy tool test for this behaviour

Concerned files
- tools/matchms/matchms_formatter.xml
- tools/matchms/formatter.py
- https://github.com/matchms/matchms/blob/master/matchms/Scores.py

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR includes the following:

- Added key parameter
- Removed formatter.py and added that code as config
- updated test to include key param

closes #318

### CODE REVIEW

File: tools/matchms/matchms_formatter.xml [hunk 1]
```diff
@@ -29,15 +28,26 @@
...
+scores = scores_from_json('$scores')
+data = []
+
+for i, (row, col) in enumerate(zip(scores.scores.row, scores.scores.col)):
+        data.append([scores.queries[col].metadata['$key'], scores.references[row].metadata['$key'], *scores.scores.data[i]])
+
+result = DataFrame(data, columns=['query', 'reference', *scores.scores.score_names])
+
+result.to_csv('$output', sep="\t", index=False)
+
```
Comment (hechth) [inline tools/matchms/matchms_formatter.xml:44]: Yeah the code has becpme quite minimal for the formatting by now, so probably a good call to move the python code into a configfile.

**Review by hechth — APPROVED** [review]
